backend/services/mgmt/v1alpha1/user-account-service/billing.go

```diff
@@ -170,21 +170,27 @@ func (s *Service) IsAccountStatusValid(
 		allowed := accountStatusResp.Msg.GetAllowedRecordCount()
 
 		if currentUsed >= allowed {
-			reason := fmt.Sprintf("Current used record count (%d) exceeds the allowed limit (%d).", currentUsed, allowed)
+			description := fmt.Sprintf("Current used record count (%d) exceeds the allowed limit (%d).", currentUsed, allowed)
 			return connect.NewResponse(&mgmtv1alpha1.IsAccountStatusValidResponse{
-				IsValid: false,
-				Reason:  &reason,
+				IsValid:            false,
+				Reason:             &description,
+				AccountStatus:      mgmtv1alpha1.AccountStatus_ACCOUNT_STATUS_EXCEEDS_ALLOWED_LIMIT.Enum(),
+				UsedRecordCount:    &currentUsed,
+				AllowedRecordCount: &allowed,
 			}), nil
 		}
 
 		if req.Msg.GetRequestedRecordCount() > 0 {
 			requested := req.Msg.GetRequestedRecordCount()
 			totalUsed := currentUsed + requested
 			if totalUsed > allowed {
-				reason := fmt.Sprintf("Adding requested record count (%d) would exceed the allowed limit (%d). Current used: %d.", requested, allowed, currentUsed)
+				description := fmt.Sprintf("Adding requested record count (%d) would exceed the allowed limit (%d). Current used: %d.", requested, allowed, currentUsed)
 				return connect.NewResponse(&mgmtv1alpha1.IsAccountStatusValidResponse{
-					IsValid: false,
-					Reason:  &reason,
+					IsValid:            false,
+					AccountStatus:      mgmtv1alpha1.AccountStatus_ACCOUNT_STATUS_REQUESTED_EXCEEDS_LIMIT.Enum(),
```

Comment:
Hm, I'm not sure we want to call `.Enum()` here, that shouldn't be necessary.

Reply:
it was only necessary since it was optional but now that it;s required we can just return back the value